src/components/structures/RightPanel.tsx

```diff
@@ -196,6 +197,15 @@ export default class RightPanel extends React.Component<IProps, IState> {
     };
 
     private onAction = (payload: ActionPayload) => {
+        const isChangingRoom = payload.action === 'view_room';
```

Comment:
view_room is also fired for permalinks and such within the same room - will this affect the code at all?

Reply:
You were spot on! Totally forgot that highlighting an event was using the same action.

Cheers for the heads up